From 59188d81cf8936cd9f5400df040d875427251bf2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Mon, 1 Oct 2018 18:05:07 +0200
Subject: [PATCH 2/2] vnc-pipewire-stream: Close session when disconnected

When there is an active stream, and we're disconnected from PipeWire
(e.g. because it terminated), close the session.
---
 src/grd-vnc-pipewire-stream.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/grd-vnc-pipewire-stream.c b/src/grd-vnc-pipewire-stream.c
index 66d66a0..d6454b8 100644
--- a/src/grd-vnc-pipewire-stream.c
+++ b/src/grd-vnc-pipewire-stream.c
@@ -392,6 +392,9 @@ on_state_changed (void                 *user_data,
         }
       break;
     case PW_REMOTE_STATE_UNCONNECTED:
+      if (stream->pipewire_stream)
+        g_signal_emit (stream, signals[CLOSED], 0);
+      break;
     case PW_REMOTE_STATE_CONNECTING:
       break;
     }
-- 
2.17.1

